main.go

```diff
@@ -308,3 +326,42 @@ func errorf(f string, a ...interface{}) {
 	}
 	exitCode = 1
 }
+
+func ensureGoModFile(m *moduleVCSInfo) error {
+	modFile := filepath.Join(m.dir, "go.mod")
+	if _, err := os.Stat(modFile); err == nil {
+		return nil
+	}
+	if err := ioutil.WriteFile(modFile, []byte(autoGoMod(m.module.Path)), 0666); err != nil {
+		return errors.Wrap(err)
+	}
+	return nil
+}
+
+// removeAutoGoMod removes the module directory's go.mod
+// file if it looks like it's been autogenerated by us.
+// It reports whether the file was removed.
+func removeAutoGoMod(m *moduleVCSInfo) (bool, error) {
+	modFile := filepath.Join(m.dir, "go.mod")
+	data, err := ioutil.ReadFile(modFile)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return false, errors.Wrap(err)
+		}
+		return false, nil
+	}
+	if string(data) != autoGoMod(m.module.Path) {
+		return false, nil
+	}
+	if err := os.Remove(modFile); err != nil {
+		return false, errors.Wrap(err)
+	}
+	return true, nil
+}
+
+// autoGoMod returns the contents of the go.mod file that
+// would be auto-generated for the module with the given
+// path.
+func autoGoMod(mpath string) string {
+	return "# autogenerated by gohack\nmodule " + mpath + "\n"
```

Comment:
Perhaps add `; DO NOT EDIT.` to the first line, to match the agreed upon pattern for generated Go source files:

```
`^// Code generated .* DO NOT EDIT\.$`
```

I don't think adding the "code generated by ..." bit is useful though, since this isn't code.

Reply:
done